doc/YamlConfigFile.md

```diff
@@ -0,0 +1,299 @@
+
+```yaml
+id: jenkins
+name: Jenkins
+description: This service runs Jenkins continuous integration system.
+env:
+    -
+        name: JENKINS_HOME
+        value: '%BASE%'
+executable: java
+arguments: >
+    -Xrs 
+    -Xmx256m 
+    -jar "%BASE%\jenkins.war" 
+    --httpPort=8080
+log:
+    mode: roll
+```
+
+## Environment variable expansion
+
+Configuration YAML files can include environment variable expansions of the form `%Name%`. 
+Such occurrences, if found, will be automatically replaced by the actual values of the variables. [Read more. . . ](xmlConfigFile.md#environment-variable-expansion)
+
+## Configuration entries
+
+### id
+
+Specifies the ID that Windows uses internally to identify the service.
+This has to be unique among all the services installed in a system, and it should consist entirely out of alpha-numeric characters.
+
+### name
+
+Short display name of the service, which can contain spaces and other characters.
+This shouldn't be too long, like `id`, and this also needs to be unique among all the services in a given system.
+
+### description
+
+Long human-readable description of the service.
+This gets displayed in Windows service manager when the service is selected.
+
+### executable
+
+This element specifies the executable to be launched.
+It can be either absolute path, or you can just specify the executable name and let it be searched from `PATH` (although note that the services often run in a different user account and therefore it might have different `PATH` than your shell does.)
+
+### startmode
+
+This element specifies the start mode of the Windows service.
+It can be one of the following values: Boot, System, Automatic, or Manual.
+For more information, see the [ChangeStartMode method](https://docs.microsoft.com/windows/win32/cimwin32prov/changestartmode-method-in-class-win32-service).
+The default value is `Automatic`.
+
+### delayedAutoStart
+
+This Boolean option enables the delayed start mode if the `Automatic` start mode is defined. [Read more. . .](xmlConfigFile.md#delayedautostart)
+
+```yaml
+delayedAutoStart: false
+```
+
+### depend
+Specify IDs of other services that this service depends on. 
+When service `X` depends on service `Y`, `X` can only run if `Y` is running.
+
+YAML list can be used to specify multiple dependencies.
+
+```yaml
+depend:
+    - Eventlog
+    - W32Time
+```
+
+### log
+
+Optionally set a different logging directory with `logpath` and startup `mode`: append (default), reset (clear log), ignore, roll (move to `\*.old`).
+
+User can specify all log configurations as a single YAML dictionary
+
+```yaml
+log:
+    mode: roll-by-size
+    logpath: '%BASE/log%'
+    sizeThreshold: 10240
+    keepFiles: 8
+```
+
+See the [Logging and error reporting](loggingAndErrorReporting.md) page for more info.
+
+### Arguments
+
+`arguments` element specifies the arguments to be passed to the executable. User can specify all the commands as a single line.
+
+```yaml
+args: arg1 arg2 arg3
```

Comment:
As per the discussion in Gitter we don't want abbreviations. Can you confirm @oleg-nenashev?

Reply:
Updated